packages/playwright-test/src/matchers/toMatchSnapshot.ts

```diff
@@ -66,3 +70,17 @@ export function toMatchSnapshot(this: ReturnType<Expect['getState']>, received:
     testInfo.attachments.push({ name: 'diff', contentType, path: diffPath });
   return { pass, message: () => message || '' };
 }
+
+export function parseToMatchSnapshotOptions(
+  nameOrOptions: NameOrSegments | ToMatchSnapshotOptions,
+  optOptions: Omit<ToMatchSnapshotOptions, 'name'> = {},
+): ToMatchSnapshotOptions {
+  let options: ToMatchSnapshotOptions;
+  if (Array.isArray(nameOrOptions) || typeof nameOrOptions === 'string')
+    options = { name: nameOrOptions, ...optOptions };
+  else
+    options = { ...nameOrOptions };
+  if (!options.name)
+    throw new Error(`toMatchSnapshot() requires a "name" parameter`);
```

Comment:
this error is misguiding a little for `toHaveScreenshot()` method, since there's no `toMatchSnapshot`.

I'd refrain from extracting this opt-parsing logic and just copy the whole thing to `toHaveScreenshot`